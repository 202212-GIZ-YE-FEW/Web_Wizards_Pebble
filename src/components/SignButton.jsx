import { Loading } from "react-flatifycss";

const SignButton = (props) => {
    const { text, loading } = props;
    const isLoading = loading ? "" : "hidden";
    return (
        <div className='px-3 flex gap-x-4'>
            <button className='w-full lg:w-auto bg-primary-200 focus:outline-none text-white font-semibold rounded-lg text-md px-5 py-3 '>
                {text}
            </button>
            <Loading spinner size='lg' theme='orange' className={isLoading} />
        </div>
    );
};
export default SignButton;
