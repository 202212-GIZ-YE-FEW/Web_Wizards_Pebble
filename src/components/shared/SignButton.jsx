import { Loading } from "react-flatifycss";

const SignButton = (props) => {
    const { text, loading } = props;
    const isLoading = loading ? "" : "hidden";
    return (
        <button className='w-full lg:w-auto bg-primary-200 focus:outline-none text-white font-semibold rounded-lg text-md px-5 py-3 flex gap-x-2 items-center'>
            {text}
            <Loading spinner size='sm' theme='orange' className={isLoading} />
        </button>
    );
};
export default SignButton;
