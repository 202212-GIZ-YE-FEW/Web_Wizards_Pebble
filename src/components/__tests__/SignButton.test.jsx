import renderer from "react-test-renderer";

import SignButton from "../shared/SignButton";

it("renders correctly", () => {
    const tree = renderer.create(<SignButton />).toJSON();
    expect(tree).toMatchSnapshot();
});
