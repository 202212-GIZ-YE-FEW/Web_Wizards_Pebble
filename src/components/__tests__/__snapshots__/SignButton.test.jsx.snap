// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="w-full lg:w-auto bg-primary-200 focus:outline-none text-white font-semibold rounded-lg text-md px-5 py-3 flex gap-x-2 items-center"
>
  <span
    aria-hidden={true}
    className="loading__LoadingWrapper-sc-43dx6p-0 hEsmJL hidden size-sm style-orange spinner"
    size="sm"
  />
</button>
`;
